### DISCUSSION TITLE
HTTP parameters as trace attributes?

### DISCUSSION THREAD

**Discussion by dsmiley** [open-telemetry/community#1824] [Community Discussions]
I'm looking at the standard tracing attributes for an HTTP span -- https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/http.md
I notice there is a standard name/approach to add HTTP headers.  But I see no equivalent for HTTP *parameters*.  Is there a reason why this was omitted?  Perhaps because that starts to bleed into content/body conceptually instead of metadata that OTEL focuses on, albeit it's subjective.  If I were to do so in like-kind under the same namespace such as `http.request.param.<key>` then would that be a problem / ill-advised?

**Comment by trask** [comment]
I think it makes sense (as an opt-in similar to http headers). there's some precedence, see https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#capturing-servlet-request-parameters
